hr_attendance_autoclose/models/hr_attendance.py

```diff
@@ -39,18 +54,9 @@ def check_for_incomplete_attendances(self):
         for att in stale_attendances:
             max_hours = att.employee_id.company_id.\
                 attendance_maximum_hours_per_day
-            if max_hours and att.open_worked_hours > max_hours:
-                leave_time = datetime.strptime(
-                    att.check_in, DEFAULT_SERVER_DATETIME_FORMAT) + timedelta(
-                    hours=max_hours)
-                if reason:
-                    vals = {'check_out': leave_time.strftime(
-                        DEFAULT_SERVER_DATETIME_FORMAT),
-                        'attendance_reason_ids': [(4, reason.id)]}
-                else:
-                    vals = {'check_out': leave_time.strftime(
-                        DEFAULT_SERVER_DATETIME_FORMAT)}
-                att.write(vals)
+            close = not att.employee_id.no_autoclose
+            if close and max_hours and att.open_worked_hours > max_hours:
```

Comment:
We can put also the condition in an external method, so it is easier to extend, what do you think?

Reply:
Done @aheficent !

Reply:
great!